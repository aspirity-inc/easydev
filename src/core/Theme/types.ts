import type { ReactNode } from 'react';

export type ThemeTypeVariant = 'light' | 'dark';

export type ThemeType = {
  type: ThemeTypeVariant;
  colors: { [key: string]: unknown };
  shadows: { [key: string]: unknown };
  spacing: { [key: string]: unknown };
  typography: {
    type: { [key: string]: unknown };
    weight: { [key: number]: string };
  };
  breakpoints: { [key: string]: string };
  transition: { [key: string]: string };
  [key: string | number]: unknown;
};

export type PartialThemeType = Partial<ThemeType>;
export type ThemeContextType = {
  theme: PartialThemeType;
  toggleTheme: () => void;
};

export type ThemeProviderType = {
  themeType?: ThemeTypeVariant;
  themes?: { light?: PartialThemeType; dark?: PartialThemeType };
  children: ReactNode | ReactNode[];
  target?: HTMLElement;
  customTarget?: HTMLElement;
  enableVendorPrefixes?: boolean;
  disableCSSOMInjection?: boolean;
  defaultStyledInjection?: boolean;
};
