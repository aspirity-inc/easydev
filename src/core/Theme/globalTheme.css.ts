import { createGlobalTheme, globalStyle } from "@vanilla-extract/css";

export const vars = createGlobalTheme(':root', {
  colors: {
    primary: {
      main: "#4CE1B6",
    },
    secondary: {
      main: "#4F96E8"
    },
    surface: {
      main: "#A5A5A7"
    },
    tertiary: {
      main: "#B693FF"
    },
    success: {
      main: "#55B627"
    },
    error: {
      main: "#DE0000"
    },
    warning: {
      main: "#FF9A00"
    },
    link: {
      main: "#007AFF"
    },
    background: "#FFFFFF",
    text: "#000000",
    icon: "#676769"
  },
  shadows: {
    gray: "box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07)",
    blue: "box-shadow: 0 0px 4px 0 rgba(119, 177, 237, 0.81)",
    violet: "box-shadow: 0 0px 8px 0 rgba(129, 76, 216, 0.57)",
    violet_light: "box-shadow: 0 6px 6px 0 rgba(213, 133, 250, 0.08)"
  },
  spacing: {
    padding: {
      small: '10px',
      medium: '20px',
      large: '30px',
    },
    borderRadius: {
      default: '0',
      small: '8px',
      large: '50%',
    },
  },
  typography: {
    type: {
      primary: '"Roboto Flex", sans-serif',
      code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
    },
    weight: {
      100: "100",
      200: "200",
      300: "300",
      400: "400",
      500: "500",
      600: "600",
      700: "700",
      800: "800",
      900: "900",
      1000: "1000",
    },
  },
});

globalStyle('html, body', {
  fontFamily: vars.typography.type.primary,
  color: vars.colors.text
});
