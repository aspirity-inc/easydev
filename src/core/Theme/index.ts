export { EasydevProvider, ThemeContext } from './ThemeContext';
export { THEMES } from './themes';
export { paletteGenerator } from './helpers/paletteGenerator';
export { scrollbarStyles } from './styles/scrollbar';
