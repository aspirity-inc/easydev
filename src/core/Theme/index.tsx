import { shade, tint } from 'polished';

const paletteGenerator = (color: string) => ({
  main: color,
  50: tint(1, color),
  100: tint(0.4, color),
  200: tint(0.3, color),
  300: tint(0.2, color),
  400: tint(0.1, color),
  500: color,
  600: shade(0.1, color),
  700: shade(0.2, color),
  800: shade(0.3, color),
  900: shade(0.5, color),
});

export const lightTheme = {
  type: 'light',
  colors: {
    primary: {
      ...paletteGenerator('#4CE1B6'),
    },
    secondary: {
      ...paletteGenerator('#4F96E8'),
    },
    surface: {
      ...paletteGenerator('#A5A5A7'),
    },
    tretiary: {
      ...paletteGenerator('#B693FF'),
    },
    success: {
      ...paletteGenerator('#55B627'),
    },
    error: {
      ...paletteGenerator('#DE0000'),
    },
    warning: {
      ...paletteGenerator('#FF9A00'),
    },
    link: {
      ...paletteGenerator('#0045E6'),
    },
    background: '#FFFFFF',
    text: '#000000',
    icon: '#676769',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07)',
    blue: 'box-shadow: 0 0px 4px 0 rgba(119, 177, 237, 0.81)',
    violet: 'box-shadow: 0 0px 8px 0 rgba(129, 76, 216, 0.57)',
    violet_light: 'box-shadow: 0 6px 6px 0 rgba(213, 133, 250, 0.08)',
  },
  spacing: {
    padding: {
      small: 10,
      medium: 20,
      large: 30,
    },
    borderRadius: {
      default: 0,
      small: 4,
      medium: 8,
      large: '50%',
    },
  },
  typography: {
    type: {
      primary: '"Roboto Flex", sans-serif',
      code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
    },
    weight: {
      100: '100',
      200: '200',
      300: '300',
      400: '400',
      500: '500',
      600: '600',
      700: '700',
      800: '800',
      900: '900',
      1000: '1000',
    },
  },
  breakpoints: {
    xs: '600px',
    sm: '900px',
    md: '1200px',
    lg: '1536px',
    xl: '1920px',
  },
  transition: {
    default: '0.3s',
  },
};

export const darkTheme = {
  ...lightTheme,
  type: 'dark',
  colors: {
    primary: {
      ...paletteGenerator('#6FE7C4'),
    },
    secondary: {
      ...paletteGenerator('#00C2E2'),
    },
    surface: {
      ...paletteGenerator('#766D7E'),
    },
    tretiary: {
      ...paletteGenerator('#AD9AF9'),
    },
    success: {
      ...paletteGenerator('#8BC34A'),
    },
    error: {
      ...paletteGenerator('#F44336'),
    },
    warning: {
      ...paletteGenerator('#FF9800'),
    },
    link: {
      ...paletteGenerator('#7ECEFF'),
    },
    background: '#232329',
    text: '#EFEFEF',
    icon: '#898290',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.25)',
    blue: 'box-shadow: 0 0 6px 0 rgba(76, 197, 255, 0.65)',
    violet: 'box-shadow: 0 0 11px 0 #1F034C',
    violet_light: 'box-shadow: 0 2px 18px 0 rgba(19, 9, 33, 0.45)',
  },
};
