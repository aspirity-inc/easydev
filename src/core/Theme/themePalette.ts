//TODO Palettes are hand-created because right now there is no time to create them automatically like in figma
export const lightPalette = {
  primary: {
    50: '#D5F8EE',
    100: '#C6F6E8',
    200: '#B3F2E1',
    300: '#9AEED6',
    400: '#79E8C9',
    500: '#4CE1B6',
    600: '#25DBA7',
    700: '#1FBA8E',
    800: '#1B9E79',
    900: '#178767',
  },
  secondary: {
    50: '#D5E6FA',
    100: '#C7DEF8',
    200: '#B5D3F5',
    300: '#9CC4F2',
    400: '#7BB1EE',
    500: '#4F96E8',
    600: '#267EE3',
    700: '#1A6BC7',
    800: '#165BA9',
    900: '#134D90',
  },
  surface: {
    50: '#FFFFFF',
    100: '#FBFBFE',
    200: '#F6F6F8',
    300: '#E7E7E9',
    400: '#C3C3C5',
    500: '#A5A5A7',
    600: '#7B7B7D',
    700: '#676769',
    800: '#48484A',
    900: '#272729',
  },
  tretiary: {
    50: '#EDE5FF',
    100: '#E8DDFF',
    200: '#E0D1FF',
    300: '#D5C2FF',
    400: '#D5C2FF',
    500: '#C8AEFF',
    600: '#B693FF',
    700: '#8C57FF',
    800: '#6923FF',
    900: '#4400ED',
  },
  success: {
    50: '#DFFFD7',
    100: '#C6F2B6',
    200: '#A4DF8F',
    300: '#84CE66',
    400: '#6DC34A',
    500: '#55B627',
    600: '#4BA81E',
    700: '#3C9411',
    800: '#2F8301',
    900: '#1A6600',
  },
  error: {
    50: '#FFBBBB',
    100: '#FFA4A4',
    200: '#FF8585',
    300: '#FF5D5D',
    400: '#FF2727',
    500: '#DE0000',
    600: '#BD0000',
    700: '#A00000',
    800: '#880000',
    900: '#740000',
  },
  warning: {
    50: '#FFF4E0',
    100: '#FFE2B4',
    200: '#FFCE82',
    300: '#FEB84D',
    400: '#FDA826',
    500: '#FF9A00',
    600: '#FF8F00',
    700: '#F67E00',
    800: '#EF6F00',
    900: '#E75403',
  },
  link: {
    300: '#007AFF',
    400: '#3B61F3',
    500: '#0045E6',
  },
};

export const darkPalette = {
  primary: {
    50: '#DDF9F1',
    100: '#D2F7ED',
    200: '#C2F5E7',
    300: '#AEF2DE',
    400: '#93EDD4',
    500: '#6FE7C4',
    600: '#43E0B3',
    700: '#23D4A2',
    800: '#1EB48A',
    900: '#10453B',
  },
  secondary: {
    50: '#E3FFFF',
    100: '#B7F3FE',
    200: '#35F4FF',
    300: '#4DD5EE',
    400: '#1DCBE7',
    500: '#00C2E2',
    600: '#00B1CE',
    700: '#009DB3',
    800: '#008A9C',
    900: '#006770',
  },
  surface: {
    50: '#E6E6E6',
    100: '#CECCCF',
    200: '#B4B0B7',
    300: '#9B95A0',
    400: '#898290',
    500: '#766D7E',
    600: '#665F6E',
    700: '#524C58',
    800: '#3F3A43',
    900: '#2A282C',
  },
  tretiary: {
    50: '#ECE7FE',
    100: '#E5DFFD',
    200: '#DCD4FC',
    300: '#D1C6FC',
    400: '#C1B3FB',
    500: '#AD9AF9',
    600: '#9276F7',
    700: '#8767FF',
    800: '#724DFB',
    900: '#8B6CF1',
  },
  success: {
    50: '#F1F9E9',
    100: '#DCEEC8',
    200: '#C4E0A5',
    300: '#AED581',
    400: '#9DCC65',
    500: '#8BC34A',
    600: '#7BB241',
    700: '#699F38',
    800: '#578C30',
    900: '#346A1F',
  },
  error: {
    50: '#FFEBEE',
    100: '#FECDD2',
    200: '#ED9999',
    300: '#E47373',
    400: '#F05450',
    500: '#F44336',
    600: '#E43A35',
    700: '#D22D2F',
    800: '#C52828',
    900: '#B71C1B',
  },
  warning: {
    50: '#FFE7C2',
    100: '#FFDFAE',
    200: '#FFD493',
    300: '#FFC670',
    400: '#FFB340',
    500: '#FF9800',
    600: '#D98200',
    700: '#B86F00',
    800: '#7F540B',
    900: '#623C04',
  },
  link: {
    300: '#B6E3FF',
    400: '#9EDAFF',
    500: '#7ECEFF',
  },
};
