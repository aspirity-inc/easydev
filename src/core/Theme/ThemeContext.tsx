import { createContext, useCallback, useMemo, useState } from 'react';

import { useIsomorphicEffect } from '@hooks';
import { merge } from 'lodash';
import { StyleSheetManager, ThemeProvider as StyledThemeProvider } from 'styled-components';

import { THEMES } from './themes.ts';
import type { ThemeContextType, ThemeProviderType } from './types.ts';

export const ThemeContext = createContext<ThemeContextType | null>(null);

export const EasydevProvider = ({
  children,
  themeType = 'light',
  themes = { light: {}, dark: {} },
  target,
  customTarget,
  enableVendorPrefixes = true,
  disableCSSOMInjection,
  defaultStyledInjection = false,
}: ThemeProviderType) => {
  const mergedLightTheme = useMemo(() => {
    return merge(THEMES['light'], themes.light);
  }, [themes?.light]);

  const mergedDarkTheme = useMemo(() => {
    return merge(THEMES['dark'], themes.dark);
  }, [themes?.dark]);

  const [currentTheme, setCurrentTheme] = useState(themeType === 'dark' ? mergedDarkTheme : mergedLightTheme);

  const toggleTheme = useCallback(() => {
    if (currentTheme.type === 'light') {
      setCurrentTheme(mergedDarkTheme);
    } else {
      setCurrentTheme(mergedLightTheme);
    }
  }, [currentTheme.type, mergedDarkTheme, mergedLightTheme]);

  useIsomorphicEffect(() => {
    const head = document.querySelector('head');
    const styled = document.querySelector('[data-styled="active"]') as Element;

    if (!defaultStyledInjection) {
      if (!target) customTarget ? customTarget.after(styled) : head?.prepend(styled);
    }
  }, []);

  const themeValues = useMemo(() => {
    return { toggleTheme, theme: currentTheme };
  }, [currentTheme, toggleTheme]);

  return (
    <ThemeContext.Provider value={themeValues}>
      <StyledThemeProvider theme={currentTheme}>
        <StyleSheetManager
          enableVendorPrefixes={enableVendorPrefixes}
          disableCSSOMInjection={disableCSSOMInjection}
          target={target}
        >
          {children}
        </StyleSheetManager>
      </StyledThemeProvider>
    </ThemeContext.Provider>
  );
};
