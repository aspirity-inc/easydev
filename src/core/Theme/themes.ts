import { darkPalette, lightPalette } from '@core/Theme/themePalette';

import { ThemeType } from './types';

const lightTheme: ThemeType = {
  type: 'light',
  colors: {
    ...lightPalette,
    background: '#FFFFFF',
    text: '#000000',
    icon: '#676769',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 7px 0 rgba(0, 0, 0, 0.07)',
    blue: 'box-shadow: 0 0px 4px 0 rgba(119, 177, 237, 0.81)',
    violet: 'box-shadow: 0 0px 8px 0 rgba(129, 76, 216, 0.57)',
    violet_light: 'box-shadow: 0 6px 6px 0 rgba(213, 133, 250, 0.08)',
  },
  spacing: {
    padding: {
      small: 10,
      medium: 20,
      large: 30,
    },
    borderRadius: {
      default: 0,
      small: 4,
      medium: 8,
      large: '50%',
    },
  },
  typography: {
    type: {
      primary: '"Roboto Flex", sans-serif',
      code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
    },
    weight: {
      100: '100',
      200: '200',
      300: '300',
      400: '400',
      500: '500',
      600: '600',
      700: '700',
      800: '800',
      900: '900',
      1000: '1000',
    },
  },
  breakpoints: {
    xs: '600px',
    sm: '900px',
    md: '1200px',
    lg: '1536px',
    xl: '1920px',
  },
  transition: {
    default: '0.3s',
  },
};

const darkTheme: ThemeType = {
  ...lightTheme,
  type: 'dark',
  colors: {
    ...darkPalette,
    background: '#232329',
    text: '#EFEFEF',
    icon: '#898290',
  },
  shadows: {
    gray: 'box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.25)',
    blue: 'box-shadow: 0 0 6px 0 rgba(76, 197, 255, 0.65)',
    violet: 'box-shadow: 0 0 11px 0 #1F034C',
    violet_light: 'box-shadow: 0 2px 18px 0 rgba(19, 9, 33, 0.45)',
  },
};

export const THEMES = {
  light: lightTheme,
  dark: darkTheme,
};
