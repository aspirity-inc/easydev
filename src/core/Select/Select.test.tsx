import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { expect, test } from 'vitest';

import { EasydevProvider } from '@core/Theme';

import { OptionType, Select } from '.';

const options: OptionType[] = [
  { value: 'chocolate', label: 'Chocolate' },
  { value: 'strawberry', label: 'Strawberry' },
  { value: 'vanilla', label: 'Vanilla' },
  { value: 'chocolate1', label: 'Chocolate1' },
  { value: 'strawberry1', label: 'Strawberry1' },
  { value: 'vanilla1', label: 'Vanilla1' },
  { value: 'chocolate2', label: 'Chocolate2' },
  { value: 'strawberry2', label: 'Strawberry2', isDisabled: true },
  { value: 'vanilla2', label: 'Vanilla2', isDisabled: true },
];

test('', async () => {
  const { container } = render(
    <EasydevProvider>
      <Select options={options} placeholder="SELECT" />
    </EasydevProvider>
  );

  // expect(document.body).toMatchSnapshot();

  // const selectElement = screen.getByText('SELECT').parentNode as Element;

  // await userEvent.click(selectElement);

  const openSelectListElement = container.querySelector('.react-select__indicator') as Element;
  await userEvent.click(openSelectListElement);
  // expect(container.querySelector('.react-select__indicator')).toBeInTheDocument();
  // expect(document.body).toMatchSnapshot();

  const selectMenuElement = container.querySelector('.react-select__menu') as Element;
  expect(selectMenuElement).toBeInTheDocument();

  options.forEach((option) => {
    expect(screen.getByText(option.label)).toBeInTheDocument();
  });

  // expect(document.body).toMatchSnapshot();
  // const modalClose = screen.getByText(mockModalText).parentNode;
  // expect(modalClose).toHaveStyleRule('opacity', '0');
});
