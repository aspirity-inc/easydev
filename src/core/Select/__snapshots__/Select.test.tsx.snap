// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` 1`] = `
.c0 {
  --transitionTiming: 0.3s;
  --transition: color var(--transitionTiming),background-color var(--transitionTiming),box-shadow var(--transitionTiming),border-color var(--transitionTiming);
}

.c0 .react-select__control {
  position: relative;
  height: 56px;
  padding: 0 16px;
  font-size: 16px;
  font-weight: 125;
  font-variation-settings: 'wdth' 110,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
  background-color: #FFFFFF;
  border: 1px solid #C3C3C5;
  border-radius: 8px;
  color: #272729;
  -webkit-transition: var(--transition);
  transition: var(--transition);
  z-index: 2;
  gap: 6px;
}

.c0 .react-select__control:hover {
  cursor: pointer;
}

.c0 .react-select__control:hover,
.c0 .react-select__control:active,
.c0 .react-select__control:focus-within {
  box-shadow: 0 0px 4px 0 rgba(119, 177, 237, 0.81);
  border-color: #9CC4F2;
}

.c0 .react-select__control:has(.react-select__value-container--has-value) {
  border-color: #9CC4F2;
}

.c0 .react-select__control .icon-indicator {
  -webkit-transition: -webkit-transform var(--transitionTiming);
  transition: transform var(--transitionTiming);
  color: inherit;
}

.c0 .react-select__control--menu-is-open .icon-indicator,
.c0 .react-select__control--menu-is-open .react-select__dropdown-indicator {
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 .react-select__control--is-disabled,
.c0 .react-select__control--is-disabled:has(.react-select__value-container--has-value) {
  color: #A5A5A7;
  border: 1px solid #C3C3C5;
  pointer-events: none;
}

.c0 .react-select__placeholder {
  font-size: 16px;
  font-weight: 125;
  font-variation-settings: 'wdth' 110,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
  color: inherit;
}

.c0 .react-select__indicator-separator {
  display: none;
}

.c0 .react-select__menu {
  margin-top: 8px;
  border: 1px solid #C7DEF8;
  border-radius: 8px;
  background-color: #FFFFFF;
  overflow: hidden;
}

.c0 .react-select__menu-list {
  --scrollbarColor: #C3C3C5;
  overflow: auto;
  scrollbar-face-color: var(--scrollbarColor);
  scrollbar-face-color: var(--scrollbarColor);
  scrollbar-track-color: transparent;
  scrollbar-arrow-color: var(--scrollbarColor);
  scrollbar-shadow-color: var(--scrollbarColor);
  scrollbar-dark-shadow-color: var(--scrollbarColor);
  scrollbar-width: thin;
  scrollbar-color: var(--scrollbarColor) transparent;
  margin-right: 4px;
  -webkit-scrollbar-gutter: stable;
  -ms-scrollbar-gutter: stable;
  scrollbar-gutter: stable;
}

.c0 .react-select__menu-list::-webkit-scrollbar-track {
  border-radius: 4px;
  background-color: transparent;
  -webkit-background-clip: content-box;
  background-clip: content-box;
}

.c0 .react-select__menu-list::-webkit-scrollbar {
  width: 8px;
  height: 8px;
  background-color: #FFFFFF;
}

.c0 .react-select__menu-list::-webkit-scrollbar-thumb {
  border-radius: 4px;
  background-color: var(--scrollbarColor);
}

.c0 .react-select__menu-notice--no-options {
  padding: 0 16px;
  text-align: left;
  color: #676769;
  font-size: 12px;
  font-weight: 450;
  font-variation-settings: 'wdth' 131,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
}

.c0 .react-select__menu-notice--loading {
  padding: 16px;
  color: #676769;
  font-size: 12px;
  font-weight: 450;
  font-variation-settings: 'wdth' 131,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
}

.c0 .react-select__menu:has(.react-select__menu-notice--no-options) {
  border: 1px solid transparent;
}

.c0 .react-select__option {
  height: 56px;
  padding: 0 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  font-weight: 125;
  font-variation-settings: 'wdth' 110,'GRAD' 0,'slnt' 0,'XTRA' 468,'XOPQ' 96,'YOPQ' 79,'YTLC' 514,'YTUC' 712,'YTAS' 750,'YTDE' -203,'YTFI' 738;
  color: inherit;
  -webkit-transition: var(--transition);
  transition: var(--transition);
}

.c0 .react-select__option .icon {
  display: none;
}

.c0 .react-select__option--is-selected .icon {
  display: block;
  color: #1A6BC7;
  -webkit-transition: var(--transition);
  transition: var(--transition);
}

.c0 .react-select__option--is-disabled {
  color: #C3C3C5;
  pointer-events: none;
}

.c0 .react-select__option:hover {
  cursor: pointer;
  background-color: #F6F6F8;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="easy_select-container react-select__container css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-cp01gg-control"
        >
          <div
            class="react-select__value-container css-1qpkzkj-ValueContainer"
          >
            <div
              class="react-select__placeholder css-1vlsb4t-placeholder"
              id="react-select-2-placeholder"
            >
              SELECT
            </div>
            <div
              class="react-select__input-container css-1q5atpo-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="react-select__indicator-separator css-j4w2j1-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-g56vrd-indicatorContainer"
            >
              <div
                class="icon-indicator"
              >
                <div
                  class="material-symbols-rounded"
                >
                  arrow_drop_down
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
